src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/testhosts/BasicHttpXmlSerFormatAttrServiceHost.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ServiceModel;
+using System.ServiceModel.Activation;
+using System.ServiceModel.Channels;
+
+namespace WcfService
+{
+    public class RpcEncSingleNsServiceHostFactory : ServiceHostFactory
+    {
+        protected override ServiceHost CreateServiceHost(Type serviceType, Uri[] baseAddresses)
+        {
+            var serviceHost = new RpcEncSingleNsServiceHost(serviceType, baseAddresses);
+            return serviceHost;
+        }
+    }
+
+    public class RpcEncSingleNsServiceHost : TestServiceHostBase<ICalculator>
```

Comment:
Can we rename this to `XmlSerializerICalculatorServiceHost` and reuse it for all service implementing `TestServiceHostBase<ICalculator>`?

Reply:
Updated.